avalon/pipeline.py

```diff
@@ -926,61 +926,103 @@ def get_representation_context(representation):
     return context
 
 
-def update_current_task(task=None, asset=None, app=None):
-    """Update active Session to a new task work area.
+def compute_session_changes(session, task=None, asset=None, app=None):
```

Comment:
I've made this a new function to compute the changes to the current session required to become a new valid session. Before we only had `update_current_task` which did the same but would *always* change the active session. That computing of the required changes is now moved to this separate function.

We might want to compute up with a better name and maybe expose it in the API?

Reply:
Or not expose `compute_session_changes` in API, but add a new arg called `dry_run=False` to `update_current_task` ?

If `dry_run` set to `True`, then return early without update.

Reply:
And maybe need to add one more *internal* arg called `_changes=None` to `update_current_task`, so we can pass the `changes` that returned from `compute_session_changes` into `update_current_task` to avoid double work.

Reply:
The `dry_run` argument does sound like a decent alternative to having this in a separate function. @mkolar @mottosso any preference on this one? Separate functions like this or make one and add `dry_run` argument?

I'm not too big a fan of adding the `_changes` internal argument. It seems like a premature optimization that for now seemingly adds redundant complexity.